### PR TITLE
Remove StructuralVariant and StructuralVariantType, add names field to Variant

### PR DESCRIPTION
**Pull request by heuermh**
Still left to do (in a later pull request) is to populate the `somatic` field from VCF INFO attributes.

### CODE REVIEW

**Comment by heuermh** [comment]
This won't pass CI until https://github.com/bigdatagenomics/bdg-formats/pull/103 is merged and the new snapshot version of bdg-formats is available.

**Comment by heuermh** [comment]
Something is git-wrong with this branch.

It is supposed to have the commit from #1126 in it (bf886c93d7aae742a9974536fff2dbf784189638) but instead has the same commit with a different hash (ea53e2d3c22fe88b340a7a769e61cd9ed068fe9f).

**Comment by fnothaft** [comment]
How'd you pull the commit over from #1126? Also, when I merged #1126 into the topic branch, I did github's squash-and-merge, which (unfortunately) triggers an update to the git commit hash.

**Comment by heuermh** [comment]
I cherry picked it over originally, then l did a rebase here against the topic branch.

**Comment by fnothaft** [comment]
If you cherry pick the commit over, I wouldn't expect the hash to be the same anyways.

**Comment by jpdna** [comment]
+1
I round tripped a VCF with `vcf2adam` then `adam2vcf` and the names (rs_ids) are present in the VCF as advertised.

File: adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala [hunk 1]
```diff
@@ -313,6 +315,36 @@ private[adam] class VariantContextConverter(dict: Option[SequenceDictionary] = N
   }
 
   /**
+   * Split the htsjdk variant context ID field into an array of names.
+   *
+   * @param vc htsjdk variant context
+   * @return Returns an Option wrapping an array of names split from the htsjdk
+   *    variant context ID field
+   */
+  private def splitIds(vc: HtsjdkVariantContext): Option[java.util.List[String]] = {
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala]: Can you write some unit tests for this? Ideally, we'd have tests for just this function with a VC with no ID, one ID, more than one ID, and an end-to-end VCF-with-IDs test.

File: adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala [hunk 2]
```diff
@@ -313,6 +315,36 @@ private[adam] class VariantContextConverter(dict: Option[SequenceDictionary] = N
...
+  }
+
+  /**
+   * Join the array of variant names into a string for the htsjdk variant context ID field.
+   *
+   * @param variant variant
+   * @return Returns an Option wrapping a string for the htsjdk variant context ID field joined
+   *    from the array of variant names
+   */
+  private def joinNames(variant: Variant): Option[String] = {
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala]: Ditto for this function RE: unit tests.

**Comment by fnothaft** [comment]
LGTM other than test nit!

File: adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala (hunk 1 above)
Comment (jpdna) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala]: > Can you write some unit tests for this?

As in me Justin?  I'd be happy to work on the unit tests - would be good for me.  But if @heuermh wants to as his PR he can have first dibs.   Just let me know what you prefer Michael - and if you want me to do it, should it be a separate PR or do you want me to PR against your PR Michael?

Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala]: Oh, sorry for the confusion! My comment was directed at @heuermh, unless he'd like you to work on it, of course!

Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala]: No worries, I got it!

**Comment by heuermh** [comment]
Added unit tests, rebased, and force pushed.  Sorry about the conflicts, not sure how to resolve them.

**Comment by heuermh** [comment]
Should I register `mutable.ArrayBuffer` or modify the unit tests (where I use `Seq()`)?
```
- can write, then read in .vcf file *** FAILED ***
  org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 4.0 failed 1 times, most recent failure: Lost task 0.0 in stage 4.0 (TID 4, localhost): com.esotericsoftware.kryo.KryoException: java.lang.IllegalArgumentException: Class is not registered: scala.collection.mutable.ArrayBuffer
Note: To register this class use: kryo.register(scala.collection.mutable.ArrayBuffer.class);
```

**Comment by fnothaft** [comment]
I'd probably register it.

**Comment by heuermh** [comment]
Jenkins, retest this please.

**Comment by heuermh** [comment]
All right, this is starting to piss me off.  Why can't I replicate this test failure locally?

**Comment by heuermh** [comment]
@fnothaft sorry, I'm not sure what to do to fix this one.

`"spark.kryo.registrationRequired" -> "true"` in ADAMFunSuite but I don't see these test failures locally.

**Comment by heuermh** [comment]
Jenkins, retest this please.

**Comment by heuermh** [comment]
WTF!  Please merge this before Jenkins changes his mind.

**Comment by fnothaft** [comment]
Merged into upgrade-to-bdg-formats-0.10.0 as https://github.com/bigdatagenomics/adam/commit/97d9290f54cdfaaa0cbfdf55a3a7aa886a3c5071. I squashed down a spurious cleanup commit and rebased upgrade-to-bdg-formats-0.10.0 on master. Thanks @heuermh!

**Comment by heuermh** [comment]
Great, thank you!
